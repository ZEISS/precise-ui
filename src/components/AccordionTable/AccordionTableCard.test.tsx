import * as React from 'react';
import * as enzyme from 'enzyme';
import { AccordionTableCard } from './AccordionTableCard.part';

describe('<AccordionTableCard />', () => {
  it('should render an empty <AccordionTableCard> component', () => {
    const wrapper = enzyme.shallow(<AccordionTableCard data={[]} detailsRenderer={() => undefined} />);
    const nodes = wrapper.filter((node: any) => node.text() !== '');
    expect(nodes.exists()).toBeFalsy();
  });

  it('should not be controlled without props', () => {
    const wrapper = enzyme.shallow(<AccordionTableCard data={[]} detailsRenderer={() => <div />} />);
    expect(wrapper.state('controlled')).toBeFalsy();
  });

  it('should be controlled with selectedIndex', () => {
    const wrapper = enzyme.shallow(<AccordionTableCard selectedIndex={0} data={[]} detailsRenderer={() => <div />} />);
    expect(wrapper.state('controlled')).toBeTruthy();
  });

  it('should not be controlled with onChange', () => {
    const wrapper = enzyme.shallow(
      <AccordionTableCard onChange={() => {}} data={[]} detailsRenderer={() => <div />} />,
    );
    expect(wrapper.state('controlled')).toBeFalsy();
  });

  it('should render a <AccordionTableCard> component with right columns', () => {
    const wrapper = enzyme.shallow(
      <AccordionTableCard data={[{ c: 5 }, { c: 10 }]} columns={{ c: 'C' }} detailsRenderer={() => <div />} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccordionTableCard> component with wrong columns', () => {
    const wrapper = enzyme.shallow(
      <AccordionTableCard data={[{ c: 5 }, { c: 10 }]} columns={{ a: 'A', b: 'B' }} detailsRenderer={() => <div />} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccordionTableCard> component with sorted data', () => {
    const wrapper = enzyme.shallow(
      <AccordionTableCard data={[{ c: 15 }, { c: 10 }]} sortBy="c" detailsRenderer={() => <div />} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccordionTableCard> component without data and custom placeholder', () => {
    const wrapper = enzyme.shallow(
      <AccordionTableCard data={[]} placeholder="No data" detailsRenderer={() => <div />} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccordionTableCard> component with data set', () => {
    const data = [
      { tag: 'C', value: 'Charlie', team: 'Charlie team' },
      { tag: 'D', value: 'Delta', team: 'Delta team' },
      { tag: 'E', value: 'Echo', team: 'Echo team' },
    ];

    const detailsRenderer = () => <div>Test details renderer</div>;

    const wrapper = enzyme.shallow(<AccordionTableCard data={data} detailsRenderer={detailsRenderer} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccordionTableCard> component with data and selectedIndex set', () => {
    const data = [
      { tag: 'A', value: 'Alpha', team: 'Alpha team' },
      { tag: 'B', value: 'Bravo', team: 'Bravo team' },
      { tag: 'C', value: 'Charlie', team: 'Charlie team' },
      { tag: 'D', value: 'Delta', team: 'Delta team' },
    ];

    const detailsRenderer = () => <div>Test details renderer</div>;

    const wrapper = enzyme.shallow(
      <AccordionTableCard selectedIndex={1} data={data} detailsRenderer={detailsRenderer} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a <AccordionTable> component with no data and placeholder set', () => {
    const detailsRenderer = () => <div>Test details renderer</div>;

    const wrapper = enzyme.shallow(
      <AccordionTableCard placeholder="No data" data={[]} detailsRenderer={detailsRenderer} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
