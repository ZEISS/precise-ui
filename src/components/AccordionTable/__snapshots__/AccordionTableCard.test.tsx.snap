// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccordionTableCard /> should render a <AccordionTable> component with no data and placeholder set 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <styled.div>
      No data
    </styled.div>
  </Styled(ListItem)>
</AccordionTableCardHost>
`;

exports[`<AccordionTableCard /> should render a <AccordionTableCard> component with data and selectedIndex set 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              A
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Alpha
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Alpha team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="1"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              B
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Bravo
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Bravo team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={true}
    >
      <div>
        Test details renderer
      </div>
    </AccordionCard>
  </Styled(ListItem)>
  <Styled(ListItem)
    key="2"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              C
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Charlie
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Charlie team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="3"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              D
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Delta
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Delta team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
</AccordionTableCardHost>
`;

exports[`<AccordionTableCard /> should render a <AccordionTableCard> component with data set 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              C
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Charlie
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Charlie team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="1"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              D
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Delta
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Delta team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="2"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              tag
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              E
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              value
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Echo
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              team
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              Echo team
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
</AccordionTableCardHost>
`;

exports[`<AccordionTableCard /> should render a <AccordionTableCard> component with right columns 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              C
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              5
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="1"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              C
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              10
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
</AccordionTableCardHost>
`;

exports[`<AccordionTableCard /> should render a <AccordionTableCard> component with sorted data 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              c
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              15
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="1"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              c
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              10
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
</AccordionTableCardHost>
`;

exports[`<AccordionTableCard /> should render a <AccordionTableCard> component with wrong columns 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              A
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              B
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
  <Styled(ListItem)
    key="1"
  >
    <AccordionCard
      header={
        Array [
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              A
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
          <ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              B
            </ForwardRef(styled.div)>
            <ForwardRef(styled.div)>
              
            </ForwardRef(styled.div)>
          </ForwardRef(styled.div)>,
        ]
      }
      onActionClick={[Function]}
      opened={false}
    />
  </Styled(ListItem)>
</AccordionTableCardHost>
`;

exports[`<AccordionTableCard /> should render a <AccordionTableCard> component without data and custom placeholder 1`] = `
<AccordionTableCardHost>
  <Styled(ListItem)
    key="0"
  >
    <styled.div>
      No data
    </styled.div>
  </Styled(ListItem)>
</AccordionTableCardHost>
`;
